### PR TITLE
fix: Select error outline style in Form

### LINKED ISSUE #27526: antd 4.8.0 LESS outputs invalid CSS
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[http://not-a-visual-bug](http://not-a-visual-bug)

### Steps to reproduce
Compile ./node_modules/antd/dist/antd.less  with lessc

### What is expected?
CSS is valid

### What is actually happening?
CSS output is invalid due to https://github.com/ant-design/ant-design/pull/27378 
The PR uses `&-focused` parent selector in LESS, but since the parent ends in a :not() pseudoselector the output becomes: 

```css
.ant-form-item-has-error .ant-select:not(.ant-select-borderless)-focused,
.ant-form-item-has-error .ant-select:not(.ant-select-borderless):focus
```

The "-focused" after :not() is illegal.

| Environment | Info |
|---|---|
| antd | 4.8.0 |
| React | 17.0.1 |
| System | macOS Catalina |
| Browser | Chrome 86 |

### PR DESCRIPTION
**Pull request by afc163**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

close #27378
close #27526 

### 💡 Background and solution

https://ant.design/components/form-cn/#components-form-demo-validate-static

image

Select focus 状态激活校验时， box-shadow 颜色 ❌

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | 💄 Fix Select abnormal outline style in Form validation.        |
| 🇨🇳 Chinese |  再次修复 Select 组件在 Form 中触发校验时外边框样式异常的问题。      |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/form/style/status.less [hunk 1]
```diff
@@ -144,13 +144,12 @@
       &.@{ant-prefix}-select-focused .@{ant-prefix}-select-selector {
         .active(@warning-color);
       }
-      &-focused,
-      &:focus {
+      &-focused {
```
Comment (xrkffgg) [inline components/form/style/status.less:147]: error 那也要改呢

**Comment by xrkffgg** [comment]
没修复 😂

**Comment by xrkffgg** [comment]
image

**Comment by afc163** [comment]
好了。

**Pull request merged**